tests/component-tests/textToSpeech/js/TextToSpeechTests.js

```diff
@@ -184,8 +232,10 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         tts.cancel();
     };
 
-    fluid.tests.textToSpeech.testUtteranceUnattached = function (tts) {
-        jqUnit.assertUndefined("The utterance should not be attached", tts.utterance);
+    // Need to wrap jqUnit.assertUndefined because the framework will throw an error when trying to resolve
+    // {tts}.utterance after it has already been detached.
```

Comment:
This doesn't seem right - if {tts} is actually a component, {tts}.utterance will always resolve if tts.utterance doesn't through - can you check this?

Reply:
Basically this is to check that the utterance subcomponent has been destroyed. When the utterance has finished speaking, it destroys itself. (https://github.com/fluid-project/infusion/pull/962/files/4ef5356a506d318686df7d010c396a008b0d459a#diff-de473a785411ce892c45482faba91a26L151-L153). If we try to call `jqUnit.assertUndefined` using the IoC syntax, we'll get an error because `{tts}.utterance` is undefined.

Reply:
It shouldn't do - could you tell me what it says?

Reply:
uncaught exception: Error: Assertion failure - check console for more details: Failed to resolve reference {arguments}.2 - could not match context with name arguments from component { typeName: "fluid.textToSpeech.utterance" gradeNames: ["fluid.textToSpeech.utterance"] id: a2murpqs-326} at path ttsPauseResumeTestEnvironment-a2murpqs-76,tts,utterance component: [object Object]

Reply:
Wed Oct 23 2019 07:12:24 GMT-0400 (Eastern Daylight Time):   ASSERTION FAILED:  Failed to resolve reference {arguments}.2 - could not match context with name arguments from component { typeName: "fluid.textToSpeech.utterance" gradeNames: ["fluid.textToSpeech.utterance"] id: 5pzyh3zu-336} at path ttsPauseResumeTestEnvironment-5pzyh3zu-76,tts,utterance component:  
Object { typeName: "fluid.textToSpeech.utterance", id: "5pzyh3zu-336", lifecycleStatus: "constructing", destroy: fabricateDestroyMethod(), options: {…}, events: {} }
Fluid.js:301:31

Reply:
I've filed this as https://issues.fluidproject.org/browse/FLUID-6418 which affects the "old framework" and verified that it is already gone in the FLUID-6148 branch https://github.com/fluid-project/infusion/pull/824/files . Also I've merged in this branch into #824 reverting this to assertUndefined and verified everything works ok. Could you just add a reminder comment here indicating that this is done because of FLUID-6418